### PR TITLE
reportable-conditions: improve logic for null site

### PR DESCRIPTION
**Pull request by kairstenfay**
Change the `reportable-conditions notify` routine to perform additional
steps when a sample is missing a linked site.

If the site is missing from a sample encounter, look in the manifest
data and relay workbook and sheet name in the Slack notification.

### CODE REVIEW

**Comment by kairstenfay** [comment]
Right now the logic I'm employing is that manifest data (workbook, sheet) only get added in the event site is null. Thoughts?

File: lib/seattleflu/db/cli/command/reportable_conditions.py [hunk 1]
```diff
@@ -166,6 +170,13 @@ def send_slack_post_request(record: NamedTupleCursor.Record, url: str) -> reques
         "condition": record.lineage
     }
 
+    if not record.site:
+        additional_data = {
```
Comment (joverlee521) [inline lib/seattleflu/db/cli/command/reportable_conditions.py]: Do we want a higher level key like `manifest` to make it clear this is manifest data?

Comment (tsibley) [reply lib/seattleflu/db/cli/command/reportable_conditions.py]: I think that's a good idea.

File: lib/seattleflu/db/cli/command/reportable_conditions.py [hunk 2]
```diff
@@ -94,8 +94,12 @@ def webhook(suffix):
             with db.savepoint(f"reportable condition presence_absence_id {record.id}"):
                 LOG.info(f"Processing reportable condition, presence_absence_id «{record.id}»")
 
+                if not record.site:
+                    LOG.info(f"No site found for presence_absence_id «{record.id}». " +
+                        "Inferring site from manifest data.")
+
                 url = SLACK_WEBHOOK_REPORTING_GENERAL \
-                    if record.site not in childrens_sites \
+                    if record.site not in childrens_sites or record.sheet != 'SCH' \
```
Comment (tsibley) [inline lib/seattleflu/db/cli/command/reportable_conditions.py]: Unless I'm turned around myself, I'm pretty sure this logic is backwards. `or` should be `and`.

**Comment by tsibley** [comment]
I pushed a fixup commit to fix the logic error and put the `workbook` and `sheet` under a `manifest` key. Take a look?

**Comment by tsibley** [comment]
[Rethink Reportable Conditions Slack bot logic](https://trello.com/c/kWKyynNH/213-rethink-reportable-conditions-slack-bot-logic)

File: lib/seattleflu/db/cli/command/reportable_conditions.py (hunk 2 above)
Comment (kairstenfay) [reply lib/seattleflu/db/cli/command/reportable_conditions.py]: Ah, you're right. I tried to make the most minimal change, but negative logic always gets me.

**Comment by kairstenfay** [comment]
It looks good, thanks!

**Comment by tsibley** [comment]
Merged and deployed!

**Pull request merged**